app/src/main/java/com/github/se/project/model/profile/Profile.kt

```diff
@@ -4,21 +4,70 @@ import java.util.EnumSet
 
 /** Data class representing a user profile. */
 data class Profile(
-    val uid: String, // Unique string id for the profile => TODO: link to google sign-in
-    val firstName: String, // First name of the user
-    val lastName: String, // Last name of the user
-    val phoneNumber: String, // Phone number of the user
-    val role: Role, // Role of the user
-    val section: Section, // Section of the user
-    val academicLevel: AcademicLevel, // Academic level of the user
-    // TODO: update languages and subjects to avoid using EnumSet
-    val languages: EnumSet<Language> = EnumSet.of(Language.STUDENT),
+    val uid: String,
+    val firstName: String,
+    val lastName: String,
+    val phoneNumber: String,
+    val role: Role,
+    val section: Section,
+    val academicLevel: AcademicLevel,
+    val languages: EnumSet<Language> = EnumSet.noneOf(Language::class.java),
     val subjects: EnumSet<TutoringSubject> = EnumSet.noneOf(TutoringSubject::class.java),
-    val schedule: List<List<Int>> = List(7) { List(12) { 0 } }, // Weekly schedule
-    var price: Int = 0,
-    // TODO: profile picture
+    val schedule: List<List<Int>> = List(7) { List(12) { 0 } },
+    var price: Int = 0
+    /*TODO: Add profile picture*/
 )
 
+data class ProfileUpload(
+    val uid: String,
+    val firstName: String,
+    val lastName: String,
+    val phoneNumber: String,
+    val role: String,
+    val section: String,
+    val academicLevel: String,
+    val languages: MutableList<String> = mutableListOf(),
+    val subjects: MutableList<String> = mutableListOf(),
+    val scheduleFlat: List<Int> = List(7 * 12) { 0 }, // Flattened version of weekly schedule
+    var price: Int = 0
+)
+
+fun profileUpload(profile: Profile): ProfileUpload {
+  return ProfileUpload(
+      uid = profile.uid,
+      firstName = profile.firstName,
+      lastName = profile.lastName,
+      phoneNumber = profile.phoneNumber,
+      role = profile.role.name,
+      section = profile.section.name,
+      academicLevel = profile.academicLevel.name,
+      languages = profile.languages.map { it.name }.toMutableList(),
+      subjects = profile.subjects.map { it.name }.toMutableList(),
+      scheduleFlat = profile.schedule.flatten(),
+      price = profile.price)
+}
+
+fun profileFromUpload(profileUpload: ProfileUpload): Profile {
+  return Profile(
+      uid = profileUpload.uid,
+      firstName = profileUpload.firstName,
+      lastName = profileUpload.lastName,
+      phoneNumber = profileUpload.phoneNumber,
+      role = Role.valueOf(profileUpload.role),
+      section = Section.valueOf(profileUpload.section),
```

Comment:
You're using valueOf() to convert strings to enums. This can cause crashes if the string does not match any enum values. Use enumValueOfOrNull to safely convert a string to an enum and handle invalid values gracefully.

Reply:
Ok, that makes sense. What should the behaviour be if that function returns null?